### PR TITLE
CANS-421: Display Modal Pop-up window 5 minutes prior to session timeout. 

### PR DESCRIPTION
**Pull request by DmitriyRudenko**
## Tests

- [x] I used TDD to develop the feature
- [x] I have included unit tests
- [ ] I have included new acceptance tests or modified existing ones
- [ ] I have included other tests

## Types of changes

- [x] New feature (non-breaking change which adds functionality)
- [ ] Refactoring (No behavioral changes)
- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

## Checklist:

- [x] My code follows the code style of this project.
- [x] I ran all my tests locally which were green.
- [x] My code adds no new issues to Code Climate
- [x] I ran all the linters for the project.
- [x] My code is in a stable state ready to be deployable, but not necessarily complete.
- [x] I promise on my honor as a CWDS developer to ensure I don't break the build (or bring donuts if I leave things broken), to check linters, use best practices, to leave the code in better shape than I found it, and to have a good day.

### CODE REVIEW

File: spec/controllers/api/security_controller_spec.rb [hunk 1]
```diff
@@ -28,5 +28,13 @@ module Api
         expect(response.status).to eq 200
       end
     end
+
+    describe '#refresh' do
+      it 'returns permisson check result' do
```
Comment (denys-davydov) [inline spec/controllers/api/security_controller_spec.rb]: Not sure it is a valid name for the test case

Comment (edgibbs) [reply spec/controllers/api/security_controller_spec.rb]: +1

Comment (DmitriyRudenko) [reply spec/controllers/api/security_controller_spec.rb]: done

File: app/javascript/Application/components/common/TimeoutWarning.test.js [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+      SecurityService.refresh = jest.fn();
+      wrapper
+        .find(Button)
+        .at(0)
+        .simulate('click');
+      expect(SecurityService.refresh).toBeCalled();
+    });
+
+    it('"Logout" button does logout on click', () => {
+      TimeoutWarning.prototype.logout = jest.fn();
```
Comment (edgibbs) [inline app/javascript/Application/components/common/TimeoutWarning.test.js]: Can't stub out the class under test here.

File: app/javascript/Application/util/TimeoutService.test.js [hunk 3]
```diff
@@ -1,26 +1,48 @@
...
     global.setTimeout = jest.fn();
     global.location.reload = jest.fn();
+    eventBus.post = jest.fn();
     timeoutService.run();
     expect(global.location.reload).toBeCalled();
     expect(global.setTimeout).not.toHaveBeenCalled();
+    expect(eventBus.post).not.toBeCalled();
   });
 
   it('timeout check is schedulled when timeout cookie is present', () => {
```
Comment (edgibbs) [inline app/javascript/Application/util/TimeoutService.test.js]: Could fix the minor schedulled typo

**Pull request merged**